### PR TITLE
TJS-22: break up large result sets by injecting newlines

### PR DESCRIPTION
**Pull request by bwknight877**
The sax parser in xmldom uses a regex to parse out newlines in the xml for whatever reason.

This `lineEndPattern` regex is causing a large enough XML payload to fail to parse.

By replacing the closing `</v>` tag with `</v>\n` with a newline, the `lineEndPattern` regex does not cause the parser to fail.

Fixes TJS-22

### CODE REVIEW

**Comment by bwknight877** [comment]
FWIW, I don't love this fix, but without putting a lot of work into fixing the sax parser and understanding why its trying to use a regex in the first place, this workaround is the best we can come up with that changes as little as possible

File: sax.js [hunk 1]
```diff
@@ -32,7 +32,10 @@ XMLReader.prototype = {
 		domBuilder.endDocument();
 	}
 }
-function parse(source,defaultNSMapCopy,entityMap,domBuilder,errorHandler){
+function parse(source, defaultNSMapCopy, entityMap, domBuilder, errorHandler) {
+	// inject newlines into the closing </v> tag to break up parsing of huge result sets inside the CDATA section in <ResultXML>
+	source = source.replace(new RegExp('</v>', 'g'), '</v>\n');
```
Comment (tanium-kdp) [inline sax.js]: DoN: Little shorter / more readable (?) to do `source.replace(/<v\/>/g, '</v>\n')`.

Comment (bwknight877) [reply sax.js]: I would typically agree, but escaping the forward slash `/` and with the forward slashes around the inline regex definition, it felt more clear to use `new RegExp()` instead

**Comment by connyay** [comment]
This is insanely specific to one response body. Fix doesn't belong in this lib.

**Pull request closed without merging**